core/primitives-core/src/gas.rs

```diff
@@ -0,0 +1,245 @@
+use near_gas::NearGas;
+
+/// Wrapper type over [`near_gas::NearGas`], which itself is a wrapper type over u64.
+///
+/// This wrapper exists to maintain JSON RPC compatibility. While [`NearGas`]
+/// serializes to a JSON string for precision, we need to continue serializing
+/// Gas values to JSON numbers for backward compatibility with existing clients.
+///
+/// Note: [`NearGas`] deserialization already handles both JSON numbers and JSON
+/// strings, so we don't need to redefine deserialization behavior here.
+#[derive(
+    borsh::BorshDeserialize,
+    borsh::BorshSerialize,
+    derive_more::Display,
+    derive_more::FromStr,
+    serde::Deserialize,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Default,
+    Hash,
+)]
+pub struct Gas(NearGas);
+
+impl Gas {
+    /// Maximum value for Gas (u64::MAX)
+    pub const MAX: Gas = Gas::from_gas(u64::MAX);
```

Comment:
There are plenty of places where you still manually use `Gas::from_gas(u64::MAX)`, consider replacing all those with `Gas::MAX`.